pkg/fftypes/batch.go

```diff
@@ -135,19 +138,33 @@ func (ma *BatchPayload) Manifest(id *UUID) *BatchManifest {
 	return tm
 }
 
-func (b *Batch) Manifest() *BatchManifest {
-	if b == nil {
-		return nil
+func (b *BatchPersisted) GenManifest(messages []*Message, data DataArray) *BatchManifest {
+	return (&BatchPayload{
+		TX:       b.TX,
+		Messages: messages,
+		Data:     data,
+	}).Manifest(b.ID)
+}
+
+func (b *BatchPersisted) Inflight(messages []*Message, data DataArray) *Batch {
```

Comment:
Perhaps `GenInflight` or `GenBatch` to match the helper above...

Reply:
Or maybe the helper above should just be `Manifest`... not terribly picky here, but we do now have a lot of utilities for converting the various representations of a batch.

Reply:
Went for `GenInflight` 👍